README.md

```diff
@@ -0,0 +1,64 @@
+`python3 slack_integration -p conf.json -t <slackbot-token> -c <channel-id> -b <bot-id>` 
+
+### Detailed steps for a developing your custom plugin:
+#### Diplomat's architecture
+Diplomat's software consists of the following classes: 
+
+* `Class ChatServiceToBotIntegrator`: This is the main class which runs the diplomat code. 
+It loads classes (plugins) from the `feature_plugins.py` file into memory along with the user defined `conf.json`. 
+It then loads the required plugin and passes the respective configuration to the plugin. This class has 2 main functions 
+which can be overridden. All chatbots inherit from this class. 
+    1. `request_transcript_and_convert_to_message_list`: Receive transcript from your chat service (e.g. Slack, Teams)
+    and convert it into a list of messages.
+    2. `post_chatbot_interventions`: Send messages to your chat service.     
+    <br>
+    * Sample Subclass: `Class SlackToBotIntegrator`: This is a sample slackbot integration class which inherits from the ChatService class 
+defined above. The class contains APIs to receive a transcript from slack and APIs to post interventions to slack. 
+
+* `Class FeaturePlugin`: This is the super class which will be inherited by all plugins. The class extends 1 API called
+`generate_interventions`. The API takes the `chat_transcript` and `author_id` as input. The class also receives the 
+configuration parameters passed into the constructor. 
+
+    * Sample Plugin: `class OverSpeakingPlugin`:  You can find this class in `feature_plugins.py`. We see that config is passed 
+as a parameter to the `__init__` class. The plugin has a static method called `get_overspeaking_authors`. Ultimately, the plugin
+implements the `generate_interventions class.` 
+ 
+
+#### Creating the config file: 
+The configuration file is a dictionary containing names of the plugins to be used along with the parameters required 
+by the respective plugins. e.g. The OverspeakingPlugin expects the values "Message Window Size", etc. 
+As a plugin developer, you can choose what parameters you wish to source from the environment. The config file will be 
+read and passed as an argument to the constructor of the main class. 
+
+#### Setting up the SlackBot app:
+You need to do the following to setup your slackbot:  
+* Visit https://api.slack.com/apps and click on `Create New App`. Add it to your workspace.  
+* Click on `Workflow Steps` from the side bar and click on `Add features and functionality` and enable `Incoming Webhooks`.
+* Click on `OAuth & Permissions` and add scopes to your bot. Ideally, you would need `channels:history`, `chat:write`,
+`incoming-webhook`, `groups:history` to your chatbot. You can add more fields depending on your requirements. 
+* Reinstall the app to your workspace. 
+* You should be able to find the `Bot User OAuth Access Token` here. That will be the `bot token` used while running `diplomat`. 
+
+
+#### Getting the channel id / bot id:
```

Comment:
I think we should leave out the "tricky" and "hacky" terminology since this is the only method that we describe here. We can also add a sentence though, saying that you can get these values using the API.